lib/Http/UserStorage.php

```diff
@@ -30,7 +31,7 @@ public function  __construct(Session $sessionHandler, EntityManager $entityManag
 	public function setIdentity(IIdentity $identity = NULL)
 	{
 		if ($identity !== NULL) {
-			$class = get_class($identity);
+			$class = $identity instanceof Proxy ? get_parent_class($identity) : get_class($identity);
```

Comment:
It'd probably be better to use this helper method: https://github.com/doctrine/common/blob/v2.6.1/lib/Doctrine/Common/Util/ClassUtils.php#L40